csharp/ql/src/utils/model-generator/CaptureSummaryModelsSpecific.qll

```diff
@@ -0,0 +1,11 @@
+import csharp
+import semmle.code.csharp.dataflow.TaintTracking
+import semmle.code.csharp.dataflow.internal.DataFlowImplCommon
+import semmle.code.csharp.dataflow.internal.DataFlowPrivate
+import ModelGeneratorUtils
+
+predicate isOwnInstanceAccess(ReturnStmt rtn) { rtn.getExpr() instanceof ThisAccess }
```

Comment:
@hvitved : Do you agree that this is correct (simpler) implementation for C# as this _always_ refers to it's own instance, as C# unlike Java doesn't have the concept of inner classes (they still need a reference to an object of the outer class to refer to it).

Reply:
This predicate should be removed, and `isOwnInstanceAccessNode` should be used instead.